DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -622,14 +687,25 @@ export const backupAndDeleteAllUserData = ({
       notificationStatusModel,
       userDataBackup
     }),
-    TE.chainW(_ =>
-      // eslint-disable-next-line sort-keys
-      backupAndDeleteProfile({
-        authenticationLockService,
-        fiscalCode,
-        profileModel,
-        servicePreferencesModel,
-        userDataBackup
-      })
+    TE.chainW(() =>
+      pipe(
+        getLastEmailIfValidated({ fiscalCode, profileModel }),
+        TE.fold(
+          () => TE.right(void 0),
+          lastValidatedEmail =>
+            TE.right(pipe(lastValidatedEmail, O.toUndefined))
+        ),
+        TE.chain(lastValidatedEmail =>
+          backupAndDeleteProfile({
```

Comment:
A style suggestion: could we relocate `getLastEmailIfValidated` into `backupAndDeleteProfile` since it pertains to the profile?